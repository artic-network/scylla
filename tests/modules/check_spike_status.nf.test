nextflow_workflow {

    name "Test Workflow check_spike_status"
    script "modules/check_spike_status.nf"
    workflow "check_spike_status"
    tag "unit"

    test("Should produce spike report") {

        when {
            params {
                spike_ins = "12242"
            }
            workflow {
                """
                input[0] = Channel.of(
                    tuple(
                        "spikein_test",
                        "PlusPF",
                        file("${projectDir}/tests/test-data/spike_in/tmv_kraken_report.txt", checkIfExists: true)
                    )
                )
                input[1] = Channel.of(
                    tuple(
                        "spikein_test",
                        file("${projectDir}/tests/test-data/spike_in/tmv_mock.fastq", checkIfExists: true)
                    )
                )
                """
            }
        }

        then {
            assert workflow.success
            assert path("${outputDir}/spikein_test/classifications/spike_count_summary.json").exists()
            assert path("${outputDir}/spikein_test/classifications/spike_summary.json").exists()
            assert snapshot(workflow.out, path("${outputDir}/spikein_test/classifications/spike_count_summary.json")).match()

            // def spike_summary_json = path("${outputDir}/spikein_test/classifications/spike_count_summary.json").json
            // def data = spike_summary_json["Tobacco mosaic virus"]["12242"]
            // assert data.name == "Tobacco mosaic virus"
            // assert data.classified_count == 10
            // assert data.classified_percentage == 50.0
        }

    }

}
