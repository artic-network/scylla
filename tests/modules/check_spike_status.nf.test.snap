{
    "Should produce spike report": {
        "content": [
            {
                "0": [
                    [
                        "spikein_test",
                        "spikein_test:md5,8be32c3d8c3023ebd1dab44010d05282"
                    ]
                ],
                "status_ch": [
                    [
                        "spikein_test",
                        "spikein_test:md5,8be32c3d8c3023ebd1dab44010d05282"
                    ]
                ]
            },
            "spike_count_summary.json:md5,c025a548e1ec45241c661eea8c7d23f6"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-27T11:24:26.855097609"
    }
}